tests/unit/test_utils.py

```diff
@@ -0,0 +1,259 @@
+            }
+            with open(env_file, "w") as f:
+                yaml.dump(env_config, f)
+
+            result = load_context_config("dev", temp_dir)
+
+            expected = {
+                "VPC_CIDR": "10.0.0.0/16",  # from base
+                "COMMON_SETTING": "base_value",  # from base
+                "SHARED_SETTING": "from_env",  # env overrides base
+                "FQDN": "dev.example.com",  # from env
+                "ENV_SPECIFIC": "dev_value",  # from env
+            }
+
+            assert result == expected
+
+    def test_no_base_config(self):
+        """Test loading config when base.yaml doesn't exist."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            env_file = Path(temp_dir) / "dev.yaml"
+            env_config = {"FQDN": "dev.example.com"}
+
+            with open(env_file, "w") as f:
+                yaml.dump(env_config, f)
+
+            result = load_context_config("dev", temp_dir)
+            assert result == env_config
+
+    def test_empty_base_config(self):
+        """Test handling of empty base config file."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            # Create empty base config
+            base_file = Path(temp_dir) / "base.yaml"
+            with open(base_file, "w") as f:
+                f.write("")  # Empty file
+
+            # Create environment config
+            env_file = Path(temp_dir) / "dev.yaml"
+            env_config = {"FQDN": "dev.example.com"}
+            with open(env_file, "w") as f:
+                yaml.dump(env_config, f)
+
+            result = load_context_config("dev", temp_dir)
+            assert result == env_config
+
+    def test_empty_env_config(self):
+        """Test handling of empty environment config file."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            # Create base config
+            base_file = Path(temp_dir) / "base.yaml"
+            base_config = {"VPC_CIDR": "10.0.0.0/16"}
+            with open(base_file, "w") as f:
+                yaml.dump(base_config, f)
+
+            # Create empty environment config
+            env_file = Path(temp_dir) / "dev.yaml"
+            with open(env_file, "w") as f:
+                f.write("")  # Empty file
+
+            result = load_context_config("dev", temp_dir)
+            assert result == base_config
+
+    def test_custom_config_dir(self):
+        """Test using a custom config directory."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            custom_dir = Path(temp_dir) / "custom_config"
+            custom_dir.mkdir()
+
+            config_file = custom_dir / "dev.yaml"
+            config_data = {"FQDN": "custom.example.com"}
+
+            with open(config_file, "w") as f:
+                yaml.dump(config_data, f)
+
+            result = load_context_config("dev", str(custom_dir))
+            assert result == config_data
+
+    def test_yaml_precedence_over_yml(self):
```

Comment:
This looks like a repeat of test_multiple_config_files_raises_value_error at line 44